application/models/services/QuestionAttributeFetcher.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+/**
+ * Fetches question attribute definitions from the available providers
+ */
+
+class QuestionAttributeFetcher
+{
+    /** @var \Question the question where the attributes should apply */
+    private $question;
+
+    /** @var array<string,mixed> array of options to pass to the providers */
+    private $options = [];
+
+    /** @var array<QuestionAttributeProvider> array of question attribute providers */
+    private $providers = [];
+
+    public function __construct()
+    {
+        $this->providers = [
```

Comment:
Should be injected in constructor as arguments? Then they can be mocked.

Reply:
You mean an optional argument as so they can be overwritter, right?

Reply:
I mean this: https://en.wikipedia.org/wiki/Dependency_injection#Constructor_injection